specs/bellatrix/fork-choice.md

```diff
@@ -76,6 +77,102 @@ As per EIP-3675, before a post-transition block is finalized, `notify_forkchoice
+
+    # Only re-org the head_block block if it arrived later than the attestation deadline.
+    head_late = store.block_timeliness.get(head_root) is False
+
+    # Only suppress the fork choice update if we are confident that we will propose the next block.
+    parent_state_advanced = store.block_states[parent_root]
+    process_slots(parent_state_advanced, proposal_slot)
+    proposer_index = get_beacon_proposer_index(parent_state_advanced)
+    proposing_reorg_slot = validator_is_connected(proposer_index)
+
+    # Do not re-org if the chain is not finalizing with acceptable frequency.
+    proposal_epoch = compute_epoch_at_slot(proposal_slot)
+    epochs_since_finalization = proposal_epoch - store.finalized_checkpoint.epoch
+    finalization_ok = epochs_since_finalization <= REORG_MAX_EPOCHS_SINCE_FINALIZATION
+
+    # Single slot re-org.
+    parent_slot_ok = parent_block.slot + 1 == head_block.slot
+    time_into_slot = (store.time - store.genesis_time) % SECONDS_PER_SLOT
+    current_time_ok = (head_block.slot == current_slot or
+                       (proposal_slot == current_slot and
+                        time_into_slot <= SECONDS_PER_SLOT // INTERVALS_PER_SLOT // 2))
+    single_slot_reorg = parent_slot_ok and current_time_ok
+
+    # Shuffling stable.
+    shuffling_stable = proposal_slot % SLOTS_PER_EPOCH != 0
+
+    # FFG information of the new head_block will be competitive with the current head.
+    # FIXME: needs more justification/finalization information in the store
+    # ffg_competitive =
+    #    store.realized_justifications[parent_root] ==
+    #        store.realized_justifications[head_root] and
+    #    store.realized_finalizations[parent_root] ==
+    #        store.realized_finalization[head_root]
+    ffg_competitive = True
+
+    # Check the head weight only if the attestations from the head slot have already been applied.
+    # Implementations may want to do this in different ways, e.g. by advancing
+    # `store.time` early, or by counting queued attestations during the head block's slot.
+    if current_slot > head_block.slot:
+        assert store.proposer_boost_root == Root()
+        head_weight = get_latest_attesting_balance(store, head_root)
+        reorg_threshold = calculate_committee_fraction(justified_state, REORG_WEIGHT_THRESHOLD)
+        head_weak = head_weight < reorg_threshold
+    else:
+        head_weak = True
+
+    return all([head_late, proposing_reorg_slot, finalization_ok, single_slot_reorg,
+                shuffling_stable, ffg_competitive, head_weak])
+```
+
+> Note that the ordering of conditions is a suggestion only. Implementations are free to
+optimize by re-ordering the conditions from least to most expensive and by returning early if
+any of the early conditions are `False`.
+
+In case `should_override_forkchoice_update` returns `True`, a node SHOULD instead call
+`notify_forkchoice_updated` with the parameters of the parent block. Payload attributes should
```

Comment:
Here the parent block should be advanced to this slot, this should be noted in the PR since the expected withdrawals may have changed from the previous slot to this slot.

Reply:
Good catch, updated in 9ca5b5ea3a1e4635c2273047b035d6bf26c57ebe.